src/go/rpk/pkg/cli/cmd/cluster/config/import.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"reflect"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/out"
+	yaml "gopkg.in/yaml.v3"
+)
+
+func importConfig(
+	client *admin.AdminAPI,
+	filename string,
+	oldConfig admin.Config,
+	schema admin.ConfigSchema,
+	all bool,
+) (err error) {
+	readbackBytes, err := ioutil.ReadFile(filename)
+	out.MaybeDie(err, "Error reading file %s: %v", filename, err)
+	var readbackConfig admin.Config
+	err = yaml.Unmarshal(readbackBytes, &readbackConfig)
+	out.MaybeDie(err, "Error parsing edited config: %v", err)
+
+	type propertyDelta struct {
+		Property string
+		OldValue string
+		NewValue string
+	}
+	propertyDeltas := make([]propertyDelta, 0)
+
+	// Calculate deltas
+	upsert := make(map[string]interface{})
+	remove := make([]string, 0)
+	for k, v := range readbackConfig {
+		oldVal, haveOldVal := oldConfig[k]
+		if meta, ok := schema[k]; ok {
+			// For numeric types need special handling because
+			// yaml encoding will see '1' as an integer, even
+			// if it is given as the value for a floating point
+			// ('number') config property, and vice versa.
+			if meta.Type == "integer" {
+				switch vFloat := v.(type) {
+				case float64:
+					v = int(vFloat)
+				}
+
+				if oldVal != nil {
+					oldVal = int(oldVal.(float64))
+				}
+			} else if meta.Type == "number" {
+				switch x := v.(type) {
+				case int:
+					v = float64(x)
+				}
+			} else if meta.Type == "array" && meta.Items.Type == "string" {
+				if v != nil {
+					v = loadStringArray(v.([]interface{}))
+				}
+				if oldVal != nil {
+					oldVal = loadStringArray(oldVal.([]interface{}))
+				}
+			}
+
+			// For types that aren't numberic or array, pass them through as-is
+		}
+
+		if haveOldVal {
+			// If value changed, add it to list of updates
+			// DeepEqual because values can be slices
+			if !reflect.DeepEqual(oldVal, v) {
+				propertyDeltas = append(propertyDeltas, propertyDelta{k, fmt.Sprintf("%v", oldVal), fmt.Sprintf("%v", v)})
+				upsert[k] = v
+			}
+		} else {
+			// Present in input but not original config, insert
+			upsert[k] = v
+			propertyDeltas = append(propertyDeltas, propertyDelta{k, "", fmt.Sprintf("%v", v)})
+		}
+	}
+
+	for k := range oldConfig {
+		if _, found := readbackConfig[k]; !found {
+			meta, inSchema := schema[k]
+			if !inSchema {
```

Comment:
What would cause this to occur, such that the schema does not have something that the old config does?

Reply:
It's not expected, but this is written to be future-cautious, if ever we had mixed versions such that a very old redpanda was returning something outside a schema that had been queried from a much newer redpanda node.